statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1295 @@
+    -----------------------------------------------------------------
+    Intercept -0.0234   0.0318  -0.7345 0.4626 -0.0858  0.0390 0.0128
+    x1         1.0305   0.0578  17.8342 0.0000  0.9172  1.1437 0.0309
+    x2        -0.0134   0.0162  -0.8282 0.4076 -0.0451  0.0183 0.0236
+    x3        -1.0260   0.0328 -31.2706 0.0000 -1.0903 -0.9617 0.0169
+    x4        -0.0253   0.0336  -0.7520 0.4521 -0.0911  0.0406 0.0269
+    =================================================================
+"""
+
+_mice_example_2 = """
+    >>> imp = mice.MICEData(data)
+    >>> fml = 'y ~ x1 + x2 + x3 + x4'
+    >>> mice = mice.MICE(fml, sm.OLS, imp)
+    >>> results = []
+    >>> for x in mice:
+    >>>     results.append(x)
+    >>>     if len(results) > 10:
+    >>>         break
+"""
+
+class MICE(object):
+
+    __doc__ = """\
+    Multiple Imputation with Chained Equations.
+
+    This class can be used to fit most Statsmodels models to data sets
+    with missing values.
+
+    Parameters
+    ----------
+    model_formula : string
+        The model formula to be fit to the imputed data sets.
+    model_class : statsmodels model
+        The model to be fit to the imputed data sets.
+    data : MICEData instance
+        MICEData object containing the data set for which
+        missing values will be imputed
+    n_skip : int
+        The number of imputed datasets to skip between consecutive
+        imputed datasets that are used for analysis.
+    init_kwds : dict-like
+        Dictionary of keyword arguments passed to the init method
+        of the analysis model.
+    fit_kwds : dict-like
+        Dictionary of keyword arguments passed to the fit method
+        of the analysis model.
+
+    Examples
+    --------
+    Simple example using defaults:
+    %(mice_example_1)s
+
+    Example using an iterator:
+    %(mice_example_2)s
+    """ % {'mice_example_1' : _mice_example_1,
+           'mice_example_2' : _mice_example_2}
+
+
+    def __init__(self, model_formula, model_class, data, n_skip=3,
+                 init_kwds=None, fit_kwds=None):
+
+        self.model_formula = model_formula
+        self.model_class = model_class
+        self.n_skip = n_skip
+        self.data = data
+        self.results_list = []
+
+        self.init_kwds = init_kwds if init_kwds is not None else {}
+        self.fit_kwds = fit_kwds if fit_kwds is not None else {}
+
+
+    def __iter__(self):
+        return self
+
+
+    def next(self):
+        return self.__next__()
+
+
+    def __next__(self):
+
+        self.data.update_all(self.n_skip)
+        start_params = None
+        if len(self.results_list) > 0:
+            start_params = self.results_list[-1].params
+        model = self.model_class.from_formula(self.model_formula,
+                                              self.data.data,
+                                              **self.init_kwds)
+        self.fit_kwds.update({"start_params": start_params})
+        result = model.fit(**self.fit_kwds)
```

Comment:
another possible extension: 
`fit` from default start_params will be in many cases much slower than good start_params. It would be useful to get good start_parms from previous fits, however we don't have a standardized signature for start_params (not relevant in OLS and similar).
(We reuse start_params in some bootstrap methods but that's within a single model.)